### ISSUE TITLE
Default functionals for ProductSpace

### ISSUE THREAD

**Issue by aringh** [odlgroup/odl#564]
The default functionals from #246 #498 are only defined for `FnBase`. Implement these also for `ProductSpace`.

**Comment by adler-j** [comment]
This is now largely solved, close?

**Comment by aringh** [comment]
Just about a year later: to some extent, but not completely I think. To check I added a product space and a power space to the test suite of functionals. The product space fails in quite a few places. For the power space it it failed in only 3 places, 2 of which were easily fixed. However, it fails in the `GroupL1Norm`, because due to [these lines](https://github.com/odlgroup/odl/blob/master/odl/operator/tensor_ops.py#L1608-L1609) one cannot create a `PointwiseNorm` operator with a base space that is itself a power space. Is this a feature or a bug?

**Comment by aringh** [comment]
The added tests in PR #1287 shows that it works for power spaces. (except the `GroupL1Norm`. Maybe @adler-j or @kohr-h could have a quick look at the comment just above?)

However, as mentioned above I got a lot of fails when I tried on other product spaces. Some of the failures might be because we cannot use comparisons with `all_almost_equal`. Do we want to do something about it, do we expect them to work/not work, are we happy for now as is?
